api/app/Models/TestEntity.php

```diff
@@ -39,6 +39,16 @@ class TestEntity extends BaseModel {
     ];
 
     /**
+     * Get the access route for the entity.
+     *
+     * @return string
+     */
+    public function entityRoute()
+    {
+        return '/test/entities';
+    }
+
+    /**
```

Comment:
@artstorm actually update on this, I just merged your branch in and I realised that you're adding the `self` attribute on repository creation. I think this is perfect, I'll just implement the base transformer to handle the conversion to json for the model factories (for the apiary document).